jumanji/training/setup_train.py

```diff
@@ -393,20 +397,33 @@ def setup_training_state(
     params_state = agent.init_params(params_key)
 
     # Initialize environment states.
-    num_devices = jax.local_device_count()
+    num_local_devices = jax.local_device_count()
+    num_global_devices = jax.device_count()
+    num_workers = num_global_devices // num_local_devices
     reset_keys = jax.random.split(reset_key, agent.total_batch_size).reshape(
-        (num_devices, agent.batch_size_per_device, -1)
+        (
+            num_workers,
+            num_local_devices,
+            agent.total_batch_size // num_global_devices,
```

Comment:
Suggestion: could set this to a variable `batch_size_per_global_device = agent.total_batch_size // num_global_devices`?

Reply:
we use "local_batch_size" in sebulba for this  :+1:

Reply:
Done